src/components/doc-editor/doc-editor-app.tsx

```diff
@@ -0,0 +1,248 @@
+import React, { useCallback, useEffect, useState } from "react";
+import stringify from "json-stringify-pretty-compact";
+import { applySnapshot, getSnapshot, onSnapshot } from "mobx-state-tree";
+import { observer } from "mobx-react";
+
+import { defaultDocumentModel, defaultDocumentModelParts } from "./doc-editor-app-defaults";
+import { EditableDocumentContent } from "../document/editable-document-content";
+import { DocumentModelType, createDocumentModelWithEnv } from "../../models/document/document";
+import { DocumentContentSnapshotType } from "../../models/document/document-content";
+import { useCustomModal } from "../../hooks/use-custom-modal";
+import { urlParams } from "../../utilities/url-params";
+import { getAssetUrl } from "../../utilities/asset-utils";
+import { useAppConfig } from "../../hooks/use-stores";
+import { DocumentAnnotationToolbar } from "../document/document-annotation-toolbar";
+import { CanvasComponent } from "../document/canvas";
+import { DocEditorSettings } from "./doc-editor-settings";
+import { SettingsDialog } from "./settings-dialog";
+
+import "../document/document.scss";
+import "../four-up.sass";
+import "./doc-editor-app.scss";
+
+const kDocEditorDocKey = "clue-doc-editor-doc";
+
+export const DocEditorApp = observer(function DocEditorApp() {
+  const {document: documentURL, readOnly, noStorage } = urlParams;
+
+  const appConfig = useAppConfig();
+  const [document, setDocument] = useState(() => {
+    const savedDocString = noStorage ? undefined : window.sessionStorage.getItem(kDocEditorDocKey);
+    const initalDoc = savedDocString ? JSON.parse(savedDocString) : defaultDocumentModel;
+
+    return createDocumentModelWithEnv(appConfig, initalDoc);
+  });
+  const [remoteDocument] = useState(() => {
+    return createDocumentModelWithEnv(appConfig, getSnapshot(document));
+  });
+  const [fileHandle, setFileHandle] = useState<FileSystemHandle|undefined>();
+  const [sectionSnapshot, setSectionSnapshot] = useState<any>();
+  const [fileName, setFileName] = useState<string>("<no file loaded>");
```

Comment:
Could consider changing this message to `<session storage>` to make it a little more self-documenting.

Reply:
Yeah, I couldn't figure out a good name for this. I agree `<no file loaded>` was weird since the editor might be showing something from session storage. But `<session storage>` also seemed a little weird since they aren't symmetrical. The document will always be stored to session storage as soon as it is changed even if there is a file name shown here (unless the `noStorage` param is passed). When a filename is listed it won't be automatically saving to this file. 

I decided to add more status information to it which hopefully make it less confusing.
The values can be:
- `<new document>`
- `<restored from storage>`
- `<saved to storage>`
- `file: filename.json`
- `file: filename.json <modified>`

So when a file is loaded or saved it will tell you if it has been modified and not saved. If a file has been loaded or saved, the status doesn't make it clear whether the file is saved to storage. That seemed like it would be taking up too much space.